pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -148,66 +152,162 @@ function readProfileFiles(rootPath: string, defaultsPath: string, lockedPath: st
+  nativeReg.enumKeyNames(regKey).forEach((k) => {
+    if (k.toLowerCase() === 'integrations' && pathParts.length === 3 && pathParts[2] === 'WSL') {
+      specialObjectKeys.push(k);
+    } else if (k.toLowerCase() === 'extensions' && pathParts.length === 2) {
+      specialObjectKeys.push(k);
+    } else if (k.toLowerCase() === 'mutedchecks' && pathParts.length === 3 && pathParts[2] === 'diagnostics') {
+      specialObjectKeys.push(k);
+    } else if (schemaKeysLowerCase.includes(k.toLowerCase())) {
+      commonKeys.push(k);
+    } else {
+      unknownKeys.push(k);
+    }
+  });
+  if (unknownKeys.length) {
+    console.log(`Unrecognized keys in registry at \\${ pathParts.join('\\') }: [${ unknownKeys.join(', ') }]`);
+  }
 
+  // First process the nested keys, then process any values
   for (const k of commonKeys) {
     const schemaVal = schemaObj[k];
-    let regValue: any = null;
+    const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
 
-    if (typeof schemaVal === 'object') {
-      if (!Array.isArray(schemaVal)) {
-        const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
+    if (!innerKey) {
+      continue;
+    }
+    try {
+      regValue = readRegistryUsingSchema(schemaVal, innerKey, pathParts.concat([k]));
+    } finally {
+      nativeReg.closeKey(innerKey);
+    }
+    if (regValue) {
+      newObject ??= {};
+      (newObject as Record<string, any>)[k] = regValue;
+    }
+  }
+  for (const k of specialObjectKeys) {
+    const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
 
-        if (!innerKey) {
-          continue;
-        }
-        try {
-          regValue = readRegistryUsingSchema(schemaVal, innerKey);
-          if (regValue && (typeof regValue === 'object') && Object.keys(regValue).length === 0) {
-            // Ignore empty objects
-            regValue = null;
-          }
-        } finally {
-          nativeReg.closeKey(innerKey);
-        }
-      } else {
-        const multiSzValue = nativeReg.queryValueRaw(regKey, k);
+    if (innerKey === null) {
+      continue;
+    }
+    try {
+      regValue = readRegistryObject(innerKey, pathParts.concat([k]));
+    } catch (err: any) {
+      console.log(`Error getting registry object for ${ pathParts.join('/') }: `, err);
+    } finally {
+      nativeReg.closeKey(innerKey);
+    }
+    if (regValue) {
+      newObject ??= {};
+      (newObject as Record<string, any>)[k] = regValue;
+    }
+  }
+  const unknownValueNames: string[] = [];
 
-        if (multiSzValue) {
-          // Registry value can be a single-string or even a DWORD and parseMultiString will handle it.
-          const arrayValue = nativeReg.parseMultiString(multiSzValue as nativeReg.Value);
+  for (const k of nativeReg.enumValueNames(regKey)) {
+    if (schemaKeysLowerCase.includes(k.toLowerCase())) {
+      const kCorrectCase = schemaKeys.find(actualKey => actualKey.toLowerCase() === k.toLowerCase());
```

Comment:
We seem to accept keys with the incorrect case, but reject value (names) with the incorrect case? That seems very inconsistent.

We probably need a written spec to nail this down…

Reply:
I rewrote this code -- all registry names, both keys and value names, are now accepted ignoring case.